### PR TITLE
chore: fix typos

### PR DESCRIPTION
**Pull request by Mister-Hope**

### CODE REVIEW

**Comment by Mister-Hope** [comment]
I am really curious about how you can make tons of typos in just one single file.🤣

**Comment by Mister-Hope** [comment]
I have prettier built-in with my editor, so I just realize I introduces some formatting changes, but I think these changes would be fine. Just leave a message if you are not satisfied with any change.

**Comment by bernd-wechner** [comment]
Thanks heaps. Typos are the scourge of hastily developed one man shows ... Will review and merge when I have time (soon).

File: copy_with_style.js [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
- * Provides support of a copy button on a web page that copies a nominated elements true to its rendered style,
- * to the clipboard. Offers th option to include all styles with a <style> tag prefixing the elements outerHTML,
- * or alternately with a "style" attributes added to each element (the nominate element and all its children),
- * called ""inlining" styles.
+ * Provides support of a copy button on a web page that copies a nominated element with its rendered style
+ * to the clipboard.
  *
- * Inlining styles (applying them as a "style" attribute one each element) is expensive (slow) and produces more
- * data, conceivably much more data than not inlining them (providing them via a <style> tag), but produuces a copy
+ * Offers a option to include all styles with a <style> tag prefixing the elements outerHTML,
```
Comment (bernd-wechner) [inline copy_with_style.js]: Offers an option
prefixing the element's  outerHTML

File: copy_with_style.js [hunk 2]
```diff
@@ -1,13 +1,15 @@
...
+ * or alternately with a "style" attribute added to each element (the nominate element and all its children),
```
Comment (bernd-wechner) [inline copy_with_style.js]: the nominated element

File: copy_with_style.js [hunk 3]
```diff
@@ -20,44 +22,44 @@
...
 	class_ready = null;
 
-	// Element exclusion overrides and augmentations. These are all function that are called
-	// if defined and passed an HTML element as their sole argument. Thy must return true or
+	// Element exclusion overrides and augmentations. These are functions that are called
+	// if defined and passed an HTML element as their sole argument. They must return true or
 	// false, true to exclude, false to keep.
 	//
-	// the extra_ pair are in addition tod efault implentations
+	// the extra_pair are in addition to default implementations
```
Comment (bernd-wechner) [inline copy_with_style.js]: "extra_" pair

As in, it refers to extra_deep_exclusions and extra_shallow_exclusions

File: copy_with_style.js [hunk 4]
```diff
@@ -20,44 +22,44 @@
...
-	// Shallow exclusion exlcude the element and graft its chidlren onto its parent.
+	// Deep exclusions exclude the element and all its children.
+	// Shallow exclusion exclude the element and graft its children onto its parent.
 	deep_exclusions = null;
 	shallow_exclusions = null;
 	extra_deep_exclusions = null;
 	extra_shallow_exclusions = null;
 
-	// If we are inlining styles on a huuuge element we may wish to permit a responsive UI while we're inlining it.
+	// If we are inlining styles on a huge element we may wish to permit a responsive UI while we're inlining it.
```
Comment (bernd-wechner) [inline copy_with_style.js:62]: He, he, I don't mind either way, but this was a pun as in a Trumpian Huuuuge ;-), but I have a page with a huuuge element.

File: copy_with_style.js [hunk 5]
```diff
@@ -69,21 +71,21 @@ class Copy_With_Style {
 	//  a list of two ints describing [threshold, frequency]
 	// 		where we defer to UI only if there are more than threshold elements
 	//		and only after batches of "frequency" elements are inlined.
-	//      a frequency of 0 request optimisation (i.e. number of elements per progress bar pixel)
+	//      a frequency of 0 request optimization (i.e. number of elements per progress bar pixel)
```
Comment (bernd-wechner) [inline copy_with_style.js:74]: Actually we use s (English spelling, z is 'murican) But I don't really care, but it's likely I see English versions all over

File: copy_with_style.js [hunk 6]
```diff
@@ -175,33 +177,42 @@ class Copy_With_Style {
...
-		this.extra_shallow_exclusions = extra_shallow_exclusions,
-		this.debug = debug;
+		(this.class_button = class_button),
+			(this.class_preparing = class_preparing),
+			(this.class_ready = class_ready),
+			(this.deep_exclusions = deep_exclusions),
+			(this.shallow_exclusions = shallow_exclusions),
+			(this.extra_deep_exclusions = extra_deep_exclusions),
+			(this.extra_shallow_exclusions = extra_shallow_exclusions),
+			(this.debug = debug);
```
Comment (bernd-wechner) [inline copy_with_style.js]: Not sure what your editor did here or why. Looks odd, in that it selectively seems to have put some of the options in brackets? Worth undoing?

File: copy_with_style.js [hunk 7]
```diff
@@ -175,33 +177,42 @@ class Copy_With_Style {
...
 		this.log_performance = log_performance;
 		this.log_HTML_to_console = log_HTML_to_console;
 		this.log_text_to_console = log_text_to_console;
 		this.check_clone_integrity = check_clone_integrity;
 		this.classes_to_debug = classes_to_debug;
 		this.styles_to_debug = styles_to_debug;
-		this.tags_to_debug = tags_to_debug.map(name => name.toUpperCase()); // tagNames reported in JS are always Uppercase
+		this.tags_to_debug = tags_to_debug.map((name) => name.toUpperCase()); // tagNames reported in JS are always Uppercase
```
Comment (bernd-wechner) [inline copy_with_style.js:194]: I don't much care one way or the other here but am curious and cautious about consistency, vs spot changes. I use the first syntax (without brackets) everywhere. Looks cleaner to me. But if it were to change to bracketed I'd look to do it across the whole file I guess.

File: copy_with_style.js [hunk 8]
```diff
@@ -388,38 +416,57 @@ class Copy_With_Style {
...
 						defer[1] = Math.floor(nelements / bar_width);
-						if (this.debug) console.log(`Optimised UI deferral: [${defer}] by spreading ${nelements} elements over ${bar_width} pixels of progress bar.`);
+						if (this.debug)
+							console.log(
+								`Optimized UI deferral: [${defer}] by spreading ${nelements} elements over ${bar_width} pixels of progress bar.`
+							);
 					}
 				} else {
-					// If frequency is 0, use a default, empirically optimised value
+					// If frequency is 0, use a default, empirically optimized value
```
Comment (bernd-wechner) [inline copy_with_style.js:429]: Again, english vs 'murican

File: copy_with_style.js [hunk 9]
```diff
@@ -643,62 +731,88 @@ class Copy_With_Style {
...
+		);
 
 		this.#observer.observe(this.element, {
 			subtree: true,
 			childList: true,
 			attributes: true,
 			attributeOldValue: true,
 			characterData: true,
-			characterDataOldValue: true
+			characterDataOldValue: true,
```
Comment (bernd-wechner) [inline copy_with_style.js:776]: Odd change from editor I presume. Puzzles me, are trailing commas better or customary in JS on object property lists?

File: copy_with_style.js [hunk 10]
```diff
@@ -848,7 +976,7 @@ class Copy_With_Style {
 	// return the twig that an element is on,
 	// being the list of nodes from element up to the first junction
 	#twig(element) {
-		let e = element
+		let e = element;
```
Comment (bernd-wechner) [inline copy_with_style.js:979]: Good catch!

**Review by bernd-wechner — CHANGES_REQUESTED** [review]
Hey, first up, huge thanks! I'm new to Github's review methods and so just trying it out. Of course the vast majority of typos are in comments because they are hastily typed and never subject to testing and my IDE clearly didn't have a spell checker ;-). But love your contribution. 

As I'm new to this process and learning let me know how it works for you, if it's troublesome, I may just explore a merge (another learning area, merging from a PR, selectively?). 

I've been flat out on some other projects (construction, physical).

**Comment by bernd-wechner** [comment]
I presume .editorconfig is from your editor too. What editor, out of interest? Amusingly I submitted a PR to another project and did same, my editor did reformatting. I have learned now for work on other projects to turn that off which I can do on a project basis. As yes, editor changes to layout confuse PRs alas. C'est la vie, we all learn.

File: copy_with_style.js (hunk 6 above)
Comment (Mister-Hope) [reply copy_with_style.js]: Prettier did this, also the most popular linting tools, I think we can change the `,` to `;`

File: copy_with_style.js (hunk 7 above)
Comment (Mister-Hope) [reply copy_with_style.js:194]: The file is formatted by prettier once I did not realize while editing, and the casing should be constant after this action.

File: copy_with_style.js (hunk 9 above)
Comment (Mister-Hope) [reply copy_with_style.js:776]: yes, when moving property sequence up and down, the trailing comma benifits

**Comment by Mister-Hope** [comment]
> I presume .editorconfig is from your editor too. What editor, out of interest? Amusingly I submitted a PR to another project and did same, my editor did reformatting. I have learned now for work on other projects to turn that off which I can do on a project basis. As yes, editor changes to layout confuse PRs alas. C'est la vie, we all learn.

Try prettier https://prettier.io/, it produces a constant stylied result. and most style changes should look good to most people

**Comment by Mister-Hope** [comment]
Should be ready

**Comment by bernd-wechner** [comment]
Thanks, will check soon. Love your work. I sent you an email BTW ;-).

**Comment by bernd-wechner** [comment]
Merged it with just quick scan. Trusting it's good ;-). Sorry I've been so busy. Did you get an email from me?

**Pull request merged**